tdms/src/hdf5_io/hdf5_base.cpp

```diff
@@ -49,8 +50,58 @@ H5Dimension HDF5Base::shape_of(const std::string &group_name,
 }
 
 bool HDF5Base::is_ok() const {
-  // TODO: check for file health might be unnessicary given we've constructed
+  // TODO: check for file health might be unnecessary given we've constructed
   // the object.
   return file_->isHdf5(filename_);
   // return file_->isAccessible(filename_) && file_->isHdf5(filename_);
 }
+
+bool HDF5Base::flagged_MATLAB_empty(const std::string &object_path) const {
+  // Can't check anything if there's no file
+  if (file_ == nullptr) { throw std::runtime_error("No file opened"); }
+
+  // This will point to the MATLAB_empty attribute
+  H5::Attribute empty_attribute;
+
+  // Attempt to fetch the object requested
+  if (!file_->exists(object_path)) {
+    throw std::runtime_error(filename_ + " has no object " + object_path);
+  }
+  hid_t object_reference = file_->getObjId(object_path);
+
+  // The object could be a group or a dataset, so we need to account for this
+  H5I_type_t object_type = file_->getHDFObjType(object_reference);
+  if (object_type == H5I_GROUP) {
+    // Dealing with a group
+    H5::Group object = file_->openGroup(object_path);
```

Comment:
```suggestion
  // Attempt to fetch the object requested
  if (!file_->exists(object_path)) {
    throw std::runtime_error(filename_ + " has no object " + object_path);
  }
  hid_t object_reference = file_->getObjId(object_path);

  H5I_type_t object_type = file_->getHDFObjType(object_reference);
  H5::Attribute empty_attribute;  // will point to the MATLAB_empty attribute
  if (object_type == H5I_GROUP) {
      H5::Group object = file_->openGroup(object_path);
```
(Move declaration of `empty_attribute` down to just before it's used and delete some comments that describe some self-describing code.)

That is to say:
```C++
int thing_named_very_well;
if (thing_named_very_well == SOME_KIND) {
    // ...
} else if (thing_named_very_well == SOME_OTHER_KIND) {
    // ...
}
```
doesn't need a comment saying `// figure out what kind of thing and do ... accordingly`. I can already see that because your code is good.